### PR TITLE
runs refresh ux improvements

### PR DESCRIPTION
**Pull request by jacobheric**
## Description

Replace the flickering scroll refresh ux element with a refresh button that has it's own loader. Also adds a toggle to disable auto refresh. 

Also, adds a send test events button to the runs page and syncs function search query term to the url search params. 

Screenshot 2024-08-20 at 8 45 45 AM

Screenshot 2024-08-20 at 8 28 08 AM

## Type of change (choose one)
- [x] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [x] I've linked any associated issues to this PR.
- [x] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/apps/dashboard/src/components/Navigation/Logo.tsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default function Logo({ collapsed, setCollapsed }: LogoProps) {
       <div className={`flex flex-row items-center justify-start ${collapsed ? '' : 'mr-3'} `}>
         {collapsed ? (
           <div className="cursor-pointer group-hover:hidden">
-            <InngestLogoSmallBW />
+            <InngestLogoSmallBW className="text-basis" />
```
Comment (jacobheric) [inline ui/apps/dashboard/src/components/Navigation/Logo.tsx:52]: Unrelated, fixes small logo for coming dark mode in the console.

File: ui/apps/dashboard/src/components/Layout/Layout.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ export default async function Layout({ activeEnv, children }: LayoutProps) {
     <div className="fixed z-50 flex h-screen w-full flex-row justify-start overflow-y-scroll overscroll-y-none">
       <SideBar collapsed={collapsed} activeEnv={activeEnv} profile={profile} />
 
-      <div className="flex w-full flex-col">
+      <div className="flex w-full flex-col overflow-x-scroll">
```
Comment (jacobheric) [inline ui/apps/dashboard/src/components/Layout/Layout.tsx:29]: Unrelated, some defense against large runs table data blowing out the layout.

File: ui/apps/dev-server-ui/src/app/(dashboard)/functions/page.tsx [hunk 3]
```diff
@@ -117,9 +118,21 @@ export default function FunctionList() {
       desc: false,
     },
   ]);
-  const [searchInput, setSearchInput] = useState('');
-  const [globalFilter, setGlobalFilter] = useState('');
+  const searchParams = useSearchParams();
```
Comment (jacobheric) [inline ui/apps/dev-server-ui/src/app/(dashboard)/functions/page.tsx:121]: Serialize function searches to the url so they persist restarts.

File: ui/apps/dev-server-ui/src/app/(dashboard)/functions/page.tsx [hunk 4]
```diff
@@ -117,9 +118,21 @@ export default function FunctionList() {
...
+  const pathname = usePathname();
+  const { replace } = useRouter();
+  const query = searchParams.get('query');
+  const [searchInput, setSearchInput] = useState(query || '');
+  const [globalFilter, setGlobalFilter] = useState(query || '');
+
   const debouncedSearch = useDebounce(() => {
+    const params = new URLSearchParams(searchParams);
```
Comment (anafilipadealmeida) [inline ui/apps/dev-server-ui/src/app/(dashboard)/functions/page.tsx:129]: We have custom hooks to simplify the search params: https://github.com/inngest/inngest/blob/main/ui/packages/components/src/hooks/useSearchParam.ts

Comment (jacobheric) [reply ui/apps/dev-server-ui/src/app/(dashboard)/functions/page.tsx]: done.

**Review by djfarrelly — APPROVED** [review]

**Pull request merged**